flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceOperatorStreamTaskTest.java

```diff
@@ -86,21 +90,30 @@
 
     @Test
     public void testMetrics() throws Exception {
+        testMetrics(
+                SourceOperatorStreamTask::new,
+                new SourceOperatorFactory<>(
+                        new MockSource(Boundedness.BOUNDED, 1), WatermarkStrategy.noWatermarks()),
+                lessThanOrEqualTo(1_000_000d));
+    }
+
+    public static void testMetrics(
```

Comment:
nit: Could we move it to a separate class (either a common base class or e.g. `SourceTaskTestSet`)? It looks a bit fishy to call a static method from a different test class.